File: 876. Middle of the Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int length = getLength(head);
        length /= 2;
        while (length-- > 0) head = head.next;
        return head;
    }
    private int getLength(ListNode pointer) {
        int count = 0;
        while (pointer != null) {
            count++;
            pointer = pointer.next;
        }
        return count;
    }
}

File: 876. Middle of the Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func middleNode(head *ListNode) *ListNode {
    length := getLength(head)
    length /= 2
    for length > 0 {
        head = head.Next
        length--
    }
    return head
}
func getLength(pointer *ListNode) int {
    count := 0
    for pointer != nil {
        count++
        pointer = pointer.Next
    }
    return count
}


